sci-physics/pythia/pythia-8.3.09.ebuild

```diff
@@ -0,0 +1,188 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit toolchain-funcs
+
+MV=$(ver_cut 1-2)
+MY_P="${PN}${PV//./}"
+LHA_VER="6.2.1"
+
+DESCRIPTION="Lund Monte Carlo high-energy physics event generator"
+HOMEPAGE="https://pythia.org/"
+SRC_URI="https://pythia.org/download/${PN}${MV//./}/${MY_P}.tgz
+	test? ( lhapdf? (
+		https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/${LHA_VER}/CT10.tar.gz
+		https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/${LHA_VER}/MRST2007lomod.tar.gz
+		https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/${LHA_VER}/NNPDF23_nlo_as_0119_qed_mc.tar.gz
+		https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/${LHA_VER}/NNPDF23_nnlo_as_0119_qed_mc.tar.gz
+		https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/${LHA_VER}/cteq66.tar.gz
+		https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/${LHA_VER}/cteq6l1.tar.gz
+		https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/${LHA_VER}/unvalidated/MRST2004qed.tar.gz
```

Comment:
These are redirected to ../downloads/.. instead of ../archive/..:

```
RedirectedUrl: version 8.3.09: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/CT10.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/CT10.tar.gz
RedirectedUrl: version 8.3.09: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/MRST2007lomod.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/MRST2007lomod.tar.gz
RedirectedUrl: version 8.3.09: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/NNPDF23_nlo_as_0119_qed_mc.tar.gz-> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/NNPDF23_nlo_as_0119_qed_mc.tar.gz
RedirectedUrl: version 8.3.09: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/NNPDF23_nnlo_as_0119_qed_mc.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/NNPDF23_nnlo_as_0119_qed_mc.tar.gz
RedirectedUrl: version 8.3.09: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/cteq66.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/cteq66.tar.gz
RedirectedUrl: version 8.3.09: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/cteq6l1.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/cteq6l1.tar.gz
RedirectedUrl: version 8.3.09: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/unvalidated/MRST2004qed.tar.gz ->https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/unvalidated/MRST2004qed.tar.gz
RedirectedUrl: version 9999: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/CT10.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/CT10.tar.gz
RedirectedUrl: version 9999: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/MRST2007lomod.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/MRST2007lomod.tar.gz
RedirectedUrl: version 9999: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/NNPDF23_nlo_as_0119_qed_mc.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/NNPDF23_nlo_as_0119_qed_mc.tar.gz
RedirectedUrl: version 9999: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/NNPDF23_nnlo_as_0119_qed_mc.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/NNPDF23_nnlo_as_0119_qed_mc.tar.gz
RedirectedUrl: version 9999: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/cteq66.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/cteq66.tar.gz
RedirectedUrl: version 9999: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/cteq6l1.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/cteq6l1.tar.gz
RedirectedUrl: version 9999: SRC_URI: permanently redirected: https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/unvalidated/MRST2004qed.tar.gz -> https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/unvalidated/MRST2004qed.tar.gz
```

Reply:
They are all over the place. i.e.
```
https://www.hepforge.org/archive/lhapdf/pdfsets/v6.backup/6.2.1/CT10.tar.gz -> https://lhapdf.hepforge.org/downloads?f=pdfsets/v6.backup/6.2.1/CT10.tar.gz
https://www.hepforge.org/downloads/lhapdf/pdfsets/v6.backup/6.2.1/CT10.tar.gz -> https://lhapdf.hepforge.org/downloads?f=pdfsets/v6.backup/6.2.1/CT10.tar.gz
lhapdfsets.web.cern.ch/lhapdfsets/current/CT10.tar.gz -> https://lhapdfsets.web.cern.ch/current/CT10.tar.gz
```
all give the same checksum in the end. Not sure which to pick but since `LHAPDF` uses `http://lhapdfsets.web.cern.ch/lhapdfsets/current/` this would probably be the most reliable?

Reply:
`current` sounds like it is a moving target, so this is not an option as it would break the `SRC_URI` every time a new version is released. We need something that includes some version number and ideally doesn't cause a `RedirectedUrl` warning, but this might not be possible.

Reply:
Yes, I'll switch to downloads then, since the `unvalidated` path also does not exist at the cern url.

Reply:
I'll update these while I am at it:
```
 * failed fetching file: pythia-6.4.28.tar.xz, uri: https://dev.gentoo.org/~bicatali/distfiles/pythia-6.4.28.tar.xz
 * failed fetching file: lutp0613man2.pdf, uri: http://home.thep.lu.se/~torbjorn/pythia/lutp0613man2.pdf
 * failed fetching file: pythia8245.tgz, uri: http://home.thep.lu.se/~torbjorn/pythia8/pythia8245.tgz
```

Reply:
See direct links in https://pythia.org/releases/, for lhapdfsets, I think using http://lhapdfsets.web.cern.ch/lhapdfsets/current/ should be stable (see https://lhapdf.hepforge.org/#sets).